src/tiled/preferences.cpp

```diff
@@ -350,6 +350,22 @@ void Preferences::setMapRenderOrder(Map::RenderOrder mapRenderOrder)
                         mMapRenderOrder);
 }
 
+void Preferences::setTilesetScale(const QString & name, qreal value)
+{
+    QString str;
+    str += QLatin1String("TilesetEditor/TilesetScales/");
+    str += name;
```

Comment:
While testing this I noticed that for embedded tilesets this name is empty, which makes sense since they do not have their own file name. To avoid storing a shared zoom level for all embedded tilesets, I think it would make sense to fall back on the name of the tileset.